app/src/main/java/com/hedvig/app/feature/genericauth/CreateOtpAttemptUseCaseImpl.kt

```diff
@@ -6,28 +6,28 @@ import com.hedvig.app.util.apollo.QueryResult
 import com.hedvig.app.util.apollo.safeQuery
 import e
 
-class CreateOtpAttemptUseCase(
-  private val apolloClient: ApolloClient,
-) {
-  sealed class Result {
-    data class Success(
-      val id: String,
-    ) : Result()
+interface CreateOtpAttemptUseCase {
+  suspend fun invoke(email: String): CreateOtpResult
+}
 
-    object Error : Result()
-  }
+class CreateOtpAttemptUseCaseImpl(
+  private val apolloClient: ApolloClient,
+) : CreateOtpAttemptUseCase {
 
-  suspend operator fun invoke(email: String) = when (
+  override suspend fun invoke(email: String) = when (
     val result = apolloClient
       .mutation(CreateOtpAttemptMutation(email = email))
       .safeQuery()
   ) {
-    is QueryResult.Success -> {
-      Result.Success(result.data.login_createOtpAttempt)
-    }
+    is QueryResult.Success -> CreateOtpResult.Success(result.data.login_createOtpAttempt)
     is QueryResult.Error -> {
       result.message?.let { e { it } }
-      Result.Error
+      CreateOtpResult.Error
     }
   }
 }
+
+sealed class CreateOtpResult {
+  data class Success(val id: String) : CreateOtpResult()
+  object Error : CreateOtpResult()
+}
```

Comment:
+1 for the extraction, I like this style more! *Maybe* name this CreateOtpAttemptResult? I think it's good to keep the name matching and adding the Result/UseCase suffix to it to clearly indicate that they go together.

Also maybe go with a sealed interface instead? In general I like to default to interface unless we *need* it to be an abstract class for some reason.

```suggestion
sealed interface CreateOtpResult {
  data class Success(val id: String) : CreateOtpResult
  object Error : CreateOtpResult
}
```